cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -724,5 +728,99 @@ TEST(Substrait, ExtensionSetFromPlan) {
   EXPECT_EQ(decoded_add_func.name, "add");
 }
 
+TEST(Substrait, GetRecordBatchReader) {
+  const auto parquet_root = std::getenv("PARQUET_TEST_DATA");
+  std::string dir_string(parquet_root);
+  std::stringstream ss;
+  ss << dir_string << "/binary.parquet";
+  auto file_path = ss.str();
```

Comment:
`/` isn't a valid directory separator on Windows.  Maybe use something like (not 100% sure this is correct) `PlatformFilename::FromString(dir_string).Join("binary.parquet")`